textile/features.textile

```diff
@@ -1510,43 +1510,70 @@ h4. ChannelMetrics
-* @(PBE2)@ This clause has been renamed to #BPE2.
-** @(PBE2a)@ This clause has been renamed to #BPE2a.
-** @(PBE2b)@ This clause has been renamed to #BPE2b.
-** @(PBE2c)@ This clause has been renamed to #BPE2c.
-* @(BPE2)@ @BatchPresence@ contains the following attributes:
-** @(BPE2a)@ @clientId@ - identical to #TP3c
-** @(BPE2b)@ @action@ - identical to #TP3b - null if @error@ is present
-** @(BPE2c)@ @error@ - an @ErrorInfo@ object representing the failure reason for this channel - null if @action@ is present
+h4. BatchPublishSuccessResult
+
+* @(BPR1)@ Contains information about the result of successful publishes to a channel requested by a single @BatchPublishSpec@
+* @(BPR2)@ The attributes of @BatchPublishSuccessResult@ consist of:
+** @(BPR2a)@ @channel@ string - the name of the channel
+** @(BPR2b)@ @messageId@ string - a string containing the @messageId@ prefix for the published message(s)
+
+h4. BatchPublishFailureResult
+
+* @(BPF1)@ Contains information about the result of unsuccessful publishes to a channel requested by a single @BatchPublishSpec@
+* @(BPF2)@ The attributes of @BatchPublishFailureResult@ consist of:
+** @(BPF2a)@ @channel@ string - the name of the channel
+** @(BPF2b)@ @error@ @ErrorInfo@ - an @ErrorInfo@ indicating the reason the message(s) failed to publish
+
+h4. BatchPresenceSuccessResult
+
+* @(BGR1)@ Contains information about the result of a successful batch presence request for a single channel
+* @(BGR2)@ The attributes of @BatchPresenceSuccessResult@ consist of:
+** @(BGR2a)@ @channel@ string - the name of the channel
+** @(BGR2b)@ @presence@ @PresenceMessage[]@ - an array containing all members present on the channel if the request succeeded
+
+h4. BatchPresenceFailureResult
+
+* @(BGF1)@ Contains information about the result of an unsuccessful batch presence request for a single channel
+* @(BGF2)@ The attributes of @BatchPresenceFailureResult@ consist of:
+** @(BGF2a)@ @channel@ string - the name of the channel
+** @(BGF2b)@ @error@ @ErrorInfo@ - @ErrorInfo@ indicating the reason the presence request failed for the given channel, if unsuccessful
+
+h4. TokenRevocationTargetSpecifier
+
+* @(TRT1)@ Describes which tokens should be affected by a token revocation request
+* @(TRT2)@ The attributes of @TokenRevocationTargetSpecifier@ consist of:
+** @(TRT2a)@ @type@ string - the type of token revocation target specifier (eg. "clientId", "revocationKey", "channel")
+** @(TRT2b)@ @value@ string - the value of the token revocation target specifier
+ 
+h4. TokenRevocationSuccessResult
+
+* @(TRS1)@ Contains information about the result of a successful token revocation request for a single target specifier
+* @(TRS2)@ The attributes of @TokenRevocationSuccessResult@ consist of:
+** @(TRS2a)@ @target@ string - the target specifier
+** @(TRS2b)@ @appliesAt@ number - a timestamp at which the token revocation will take effect
```

Comment:
realtime returns both as numbers in milliseconds since the epoch (same as all its time apis). We might want sdks to expose / take a Time object where that's idiomatic, I'l lleave that to Owen to decide (if so it should probs be done for both consistently, and maybe consistently with whatever other apis that take times do, eg history start/end)